pynapple/core/time_series.py

```diff
@@ -974,6 +990,13 @@ def round_if_float(x):
             else:
                 return tabulate([], headers=headers) + "\n" + bottom
 
+    def __getattr__(self, name):
+        # wrap in try except to avoid infinite recursion from metadata pickling
+        try:
+            return MetadataBase.__getattr__(self, name)
+        except:
+            return super().__getattr__(name)
```

Comment:
this is a solution that avoids running the recursion
```python
        # avoid infinite recursion when pickling due to
        # self._metadata.column having attributes '__reduce__', '__reduce_ex__'
        if name in ("__getstate__", "__setstate__", "__reduce__", "__reduce_ex__"):
            raise AttributeError(name)
```

Another cleaner solution, was to reimplement `__getstate__` and `__setstate__`, as in here 

https://github.com/flatironinstitute/nemos/blob/main/src/nemos/basis.py

Reply:
I added in this solution!